src/haven/instrument/robot.py

```diff
@@ -0,0 +1,124 @@
+import asyncio
+import logging
+
+from ophyd import Component as Cpt
+from ophyd import DynamicDeviceComponent as DCpt
+from ophyd import Device, EpicsMotor, EpicsSignal
+
+from .._iconfig import load_config
+from .device import aload_devices, make_device
+
+log = logging.getLogger(__name__)
+
+def transfer_sample(num_dios: int):
+    """Create a dictionary with robot sample device definitions.
+    For use with an ophyd DynamicDeviceComponent.
+    Parameters
+    ==========
+    num_dios
+      How many samples to create.
+    """
+    sample = {}
+    sample_attrs = ['present', 'empty', 'load', 'unload','x', 'y', 'z', 'rx', 'ry', 'rz']
+    for n in range(num_dios):
+        for attr in sample_attrs:
+            sample[f"sample{n}_{attr}"] = Cpt(EpicsSignal, f":sample{n}:{attr}", labels={"transfer"})
+    return sample
+
+
+class Robot(Device):
+    # joints and position
+    i = Cpt(EpicsMotor, ":i", labels={ "joint"})
```

Comment:
Labels should be plural in line with other haven convention.

On that note, do we really need labels for all the signals?